Advanced-Audio-Driver-Java/app/src/main/java/com/tokbox/sample/advancedaudiodriver/AdvancedAudioDevice.java

```diff
@@ -1000,13 +1002,32 @@ private void unregisterBtReceiver() {
 
     private boolean isPhoneStateListenerRegistered;
 
+    //Phone state permissions are required to from Api 31. Adding this check to avoid crash.
+    private boolean hasPhoneStatePermission() {
+        if (Build.VERSION.SDK_INT >= 31) {
+            if (context.checkSelfPermission(Manifest.permission.READ_PHONE_STATE)
+                    != PackageManager.PERMISSION_GRANTED) {
+                Log.e(TAG, "Some features may not be available unless the phone permissions has been granted explicitly " +
+                        "in the App settings.");
+                return false;
+            }
+        }
+        return true;
+    }
+
     private void registerPhoneStateListener() {
         Log.d(TAG, "registerPhoneStateListener() called");
 
         if (isPhoneStateListenerRegistered) {
             return;
         }
 
+        if (!hasPhoneStatePermission()) {
+            Log.d(TAG, "No Phone State permissions. Register phoneStateListener cannot " +
```

Comment:
Why `Log.d` here and `Log.e` in `hasPhoneStatePermission()`? I would say the other way around makes more sense. But it is subjective... Feel free to ignore if you disagree.